### ISSUE TITLE
[Bug] location.replace is not replacing the current resource

### ISSUE THREAD

**Issue by ksy36** [mozilla-mobile/fenix#4003]
## Steps to reproduce
Go to https://t.co/R1nmCOjsDW

### Expected behavior
Replaces the current page with https://medium.com/@cdsmithus/functional-reactive-programming-with-reflex-and-codeworld-85495360f1b7

### Actual behavior
Replacement is not happening

### Device information
* Fenix version: 1.0.1926

As reported here:
https://github.com/webcompat/web-bugs/issues/34765

There is a script at the top of the page:
```
<script>window.opener = null; location.replace("https:\/\/link.medium.com\/u5HNJTRyXX")</script>
```

**Comment by ekager** [comment]
I'm not seeing this issue anymore it could have been fixed by either the site or a GV update. I'm tagging as QA needed

**Comment by softvision-miralobontiu** [comment]
This is fixed on the latest Nightly build 190807 (#12190609) with Nokia 6 (Android 7.1.1).

**Comment by marcorichetta** [comment]
Hi!
Some days ago I found Firefox Nightly unable to open links from Twitter.

I can see, from Firefox on my laptop, the t.co link returning something similar to what is described in this issue.
```html
<head><noscript><META http-equiv="refresh" content="0;URL=https://bit.ly/3MPuAsi"></noscript><title>https://bit.ly/3MPuAsi</title></head><script>window.opener = null; location.replace("https:\/\/bit.ly\/3MPuAsi")</script>
``` 

### Information
- Link: https://t.co/rcBZAL78UH (From this tweet https://twitter.com/alexxubyte/status/1529489416632225795?s=20&t=bt8s94K01QOsE2ywp7vXUw)
- Browser: Firefox Browser (Nightly for Developers) 
- Version: v102.0a.1
- OS: Android 11

### Extra info

**Firefox Screenshot** 
<img src="https://user-images.githubusercontent.com/19599150/170395679-0c7c91ce-4e3a-4446-9449-6e847bd7329a.png"  title="Screenshot_20220525-220117_Firefox_Nightly" width="200"/>

<details>
<summary>wget output</summary>

```bash
> wget t.co/rcBZAL78UH 
--2022-05-25 22:09:18--  http://t.co/rcBZAL78UH
Resolviendo t.co (t.co)... 104.244.42.133
Conectando con t.co (t.co)[104.244.42.133]:80... conectado.
Petición HTTP enviada, esperando respuesta... 301 Moved Permanently
Localización: https://t.co/rcBZAL78UH [siguiendo]
--2022-05-25 22:09:19--  https://t.co/rcBZAL78UH
Cargado certificado CA '/etc/ssl/certs/ca-certificates.crt'
Conectando con t.co (t.co)[104.244.42.133]:443... conectado.
Petición HTTP enviada, esperando respuesta... 301 Moved Permanently
Localización: https://bit.ly/3MPuAsi [siguiendo]
--2022-05-25 22:09:20--  https://bit.ly/3MPuAsi
Resolviendo bit.ly (bit.ly)... 67.199.248.10, 67.199.248.11
Conectando con bit.ly (bit.ly)[67.199.248.10]:443... conectado.
Petición HTTP enviada, esperando respuesta... 301 Moved Permanently
Localización: https://netflixtechblog.com/how-netflix-scales-its-api-with-graphql-federation-part-1-ae3557c187e2 [siguiendo]
--2022-05-25 22:09:20--  https://netflixtechblog.com/how-netflix-scales-its-api-with-graphql-federation-part-1-ae3557c187e2
...
```
</details>
